winpr/libwinpr/utils/collections/StreamPool.c

```diff
@@ -209,7 +209,7 @@ void StreamPool_Return(wStreamPool* pool, wStream* s)
 		new_cap = pool->aCapacity * 2;
 		new_arr = (wStream**) realloc(pool->aArray, sizeof(wStream*) * new_cap);
 		if (!new_arr)
-			return;
+			goto out_fail;
```

Comment:
IMHO we could and should always Stream_Free s if this function fails anywhere because we can assume that the reference to s is given up by the caller.